File: sem 3/DS/searching/binary.py
import random
nums = [1, 4, 6, 7, 8, 10, 11, 14, 20]

search =11
start = 0
end = len(nums) - 1

def binary(n, start, end):
    mid_point = (start + end) // 2
    print(n[start:end], mid_point)
    mid_val = n[mid_point]
    
    if search == mid_val:
        return mid_point
    if mid_val > search:
        return binary(n, start, mid_point - 1)
    elif mid_val < search:
        return binary(n, mid_point + 1, end)
    
    
print(binary(nums, start, end))

File: sem 3/DS/searching/binary.cpp
#include <iostream>
using namespace std;

int main() {
    int nums[9] = [1, 4, 6, 7, 8, 10, 11, 14, 20];
    int n = 9;
    int target = 11;
    int start = 0;
    int end = n - 1;

    while (start <= end) {
        int mid = (start + end) / 2;
        if (nums[mid] == target) {
            cout << mid << endl;
            break;
        } else if (nums[mid] < target) {
            start = mid + 1;
        } else if (nums[mid] > target) {
            end = mid - 1;
        }
    }
    
    return 0;
}